python/alb_ssl_policy.py

```diff
@@ -0,0 +1,407 @@
+  | ---------------------- | --------- | -------------------------------------------------------------------------------------- |
+
+Feature:
+  In order to: apply ssl policy to listers
+  As: a Security Officer
+  I want: to chase configuration of listener ssl policy of an application load balancer.
+Scenarios:
+  Scenario 1:
+    Given: SslPolicy is in COMPLIANT_POLICY 
+     Then: Return COMPLIANT
+  Scenario 2:
+    Given: SslPolicy is not in COMPLIANT_POLICY
+     Then: Return COMPLIANT
+  Scenario 3:
+    Given: CompliantPolicies has not been defined.
+     Then: Return NON_COMPLIANT
+  Scenario 4:
+    Given: Listner has not been used in a application load balancer.
+     Then: Return NOT_COMPLIANT
+'''
+# Python packages
+import json
+import datetime
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# for Assume Role parameter
+ASSUME_ROLE_MODE = False
+DEFAULT_RESOURCE_TYPE = ''
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+    alb_client = get_client("elbv2", event)
+
+    listeners = alb_client.describe_listeners( LoadBalancerArn = configuration_item['configuration']['loadBalancerArn'] )
+
+    if len(valid_rule_parameters) == 0:
+        return 'NON_COMPLIANT'
+
+    if 'Listeners' not in listeners.keys():
+         return 'NON_COMPLIANT'
+    else:
+        for l in listeners['Listeners']:
+            if 'SslPolicy' in l.keys():
+                if l['SslPolicy'] not in valid_rule_parameters['ValidPolicies']:
+                    return 'NON_COMPLIANT'

```

Comment:
Use build_evaluation_from_config_item() to add an annotation to the NON_COMPLIANT to explain why to the customer